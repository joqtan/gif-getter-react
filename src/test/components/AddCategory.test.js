import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory'

describe('test con <AddCategory/> component', () => {
    const setCategories = jest.fn()
    let wrapper = shallow(<AddCategory setCategories={setCategories} />)

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories={setCategories} />)
    })

    test('should it render properly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should change the input', () => {
        const value = 'hey'
        const input = wrapper.find('input')
        input.simulate('change', { target: { value } })
        expect(wrapper.find('p').text().trim()).toBe(value)
    })

    test('should not call the event with submit', () => {
        const form = wrapper.find('form')
        form.simulate('submit', { preventDefault() {} })
        expect(setCategories).not.toHaveBeenCalled()
    })

    test('should call setCategories and clean the input', () => {
        const value = 'hey'
        wrapper.find('input').simulate('change', { target: { value } })
        wrapper.find('form').simulate('submit', { preventDefault() {} })
        expect(setCategories).toHaveBeenCalled()
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function))
        expect(wrapper.find('input').prop('value')).toBe('')
    })
})
