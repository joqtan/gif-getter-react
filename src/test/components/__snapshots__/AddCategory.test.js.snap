// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test con <AddCategory/> component should it render properly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="type a new category..."
    type="text"
    value=""
  />
</form>
`;
